pkg/transform/kubernetes/kubernetes.go

```diff
@@ -217,23 +243,53 @@ func (t *Transformer) CreateDeployments(s *object.Service) ([]runtime.Object, er
 		// TODO: It is assumed that the check is done about the existence of volume in root level volume section
 		for _, mount := range c.Mounts {
 			volumeMount := api_v1.VolumeMount{
-				Name:      mount.VolumeRef,
 				ReadOnly:  mount.ReadOnly,
 				MountPath: mount.MountPath,
 				SubPath:   mount.VolumeSubPath,
 			}
 
+			if mount.VolumeRef != nil {
+				volumeMount.Name = *mount.VolumeRef
+			}
+
+			if mount.SecretRef != nil {
+				volumeMount.ReadOnly = true
+
```

Comment:
```yaml
        volumeMounts:
        - mountPath: /tmp/asdf
          name: ""
          readOnly: true
```
invalid deployment gets with volumeMoutns that look like this generated because you are not setting `volumeMount.name` here.

```go
volumeMount.Name = secretDef.SecretName
```